src/components/Scrollable/Scrollable.jsx

```diff
@@ -107,32 +109,41 @@ export default class Scrollable extends React.PureComponent {
 
     updateScrollbars = () => {
         this.event.next = this.getEvent();
-        // This check ensures that updates (which are a potentially expensive operation)
+
+        const nextEvent = this.event.next,
+            prevEvent = this.event.prev,
+            changed = nextEvent.scrollHeight !== prevEvent.scrollHeight ||
+            nextEvent.scrollWidth !== prevEvent.scrollWidth ||
+            nextEvent.top !== prevEvent.top ||
+            nextEvent.left !== prevEvent.left;
+
+        // Ensures that updates (which are a potentially expensive operation)
         // are only executed if the applicable scroll properties have changed
-        if (!isEqual(this.event.prev, this.event.next)) {
+        if (changed) {
             const el = this.container.current.parentElement;
-            this.props.onUpdate(this.event.next);
-            const vRatio = this.event.next.clientHeight / this.event.next.scrollHeight;
-            const hRatio = this.event.next.clientWidth / this.event.next.scrollWidth;
-
-            if (vRatio < 1) {
-                el.classList.add('vertically-scrollable');
-            } else {
-                el.classList.remove('vertically-scrollable');
-            }
-
-            if (hRatio < 1) {
-                el.classList.add('horizontally-scrollable');
-            } else {
-                el.classList.remove('horizontally-scrollable');
-            }
-
-            el.style.setProperty('--scrollable-vertical-ratio', vRatio);
-            el.style.setProperty('--scrollable-horizontal-ratio', hRatio);
-            el.style.setProperty('--scrollable-scroll-top', this.event.next.top);
-            el.style.setProperty('--scrollable-scroll-left', this.event.next.left);
+            this.props.onUpdate(nextEvent);
+            const vRatio = nextEvent.clientHeight / nextEvent.scrollHeight;
+            const hRatio = nextEvent.clientWidth / nextEvent.scrollWidth;
+
+            requestAnimationFrame(() => {
```

Comment:
Since the DOM is being modified directly and not by React, it is best to wrap the write operations with `requestAnimationFrame` because writing to the DOM is expensive (performance)